packages/ui-extensions/src/surfaces/customer-account/api/shared.ts

```diff
@@ -428,3 +428,175 @@ export interface SessionToken {
    */
   get(): Promise<string>;
 }
+
+export interface AllowedProcessing {
+  /**
+   * Can collect customer analytics about how the shop was used and interactions made on the shop.
+   */
+  analytics: boolean;
+  /**
+   * Can collect customer preference for marketing, attribution and targeted advertising from the merchant.
+   */
+  marketing: boolean;
+  /**
+   * Can collect customer preferences such as language, currency, size, and more.
+   */
+  preferences: boolean;
+  /**
+   * Can collect customer preference for sharing data with third parties, usually for behavioral advertising.
+   */
+  saleOfData: boolean;
+}
+
+export interface VisitorConsent {
+  /**
+   * Visitor consents to recording data to understand how customers interact with the site.
+   */
+  analytics?: boolean;
+  /**
+   * Visitor consents to ads and marketing communications based on customer interests.
+   */
+  marketing?: boolean;
+  /**
+   * Visitor consent to remembering customer preferences, such as country or language, to personalize visits to the website.
+   */
+  preferences?: boolean;
+  /**
+   * Opts the visitor out of data sharing / sales.
+   */
+  saleOfData?: boolean;
+}
+
+export interface TrackingConsentMetafield {
+  /**
+   * The name of the metafield. It must be between 3 and 30 characters in
+   * length (inclusive).
+   */
+  key: string;
+  /**
+   * The information to be stored as metadata.
+   *
+   * @example 'any string', '', or a stringified JSON object
+   */
+  value: string;
+}
+
+export interface TrackingConsentMetafieldChange {
+  /**
+   * The name of the metafield. It must be between 3 and 30 characters in
+   * length (inclusive).
+   */
+  key: string;
+  /**
+   * The information to be stored as metadata. If the value is `null`, the metafield will be deleted.
+   *
+   * @example 'any string', `null`, or a stringified JSON object
+   */
+  value: string | null;
+}
+
+export interface VisitorConsentChange extends VisitorConsent {
```

Comment:
This is all 1:1 the same as checkout, right? Could we move this into a shared file that both use?

Reply:
Yeah I agree, could be worth it to just share the same